analysis_driver/util/helper_functions.py

```diff
@@ -1,6 +1,24 @@
-# retrieving run_element_id, which differs between barcodeless and multiplexed runs
+import os
+
+
 def get_run_element_id(run_id, lane_number, barcode=None):
+    """Retrieving run_element_id, which differs between barcodeless and multiplexed runs"""
     if barcode is None:
         return '{dataset_name}_{lane}'.format(dataset_name=run_id, lane=lane_number)
     else:
         return '{dataset_name}_{lane}_{barcode}'.format(dataset_name=run_id, lane=lane_number, barcode=barcode)
+
+
+def prepend_path_to_data_files(prepend_path, data_structure):
```

Comment:
I see this has changed again.

Reply:
Yes I had to create a function to support recursion.